cogs/msg.py

```diff
@@ -162,19 +163,23 @@ async def caption(self, ctx, shorthand, *, caption):
 
         file_url = f"https://api.memegen.link/images/custom/_/{safe_caption}.gif?background={quoted_url}"
 
-        async with ctx.typing():
-            # Prevent 403 Forbidden errors
-            req = request.Request(file_url,
-                                  headers={
-                                      "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"
-                                  })
+        headers = {
+            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"
+        }
 
-            # Download a maximum size of 32 MB as it's an in-memory buffer
-            with request.urlopen(req, timeout=30) as r:
-                b = r.read(32 * 1024 * 1024)
-                file = discord.File(io.BytesIO(b), filename="meme.gif")
+        async with ctx.typing(), \
+                ClientSession(timeout=ClientTimeout(30), raise_for_status=True, headers=headers) as session, \
+                session.get(file_url) as resp:
+            # Restrict buffer size to 32MB
+            file_data = b''
+            async for chunk in resp.content.iter_chunked(512 * 1024):
+                file_data += chunk
+                if len(file_data) > 32 * 1024 * 1024:
+                    raise ValueError('File too large')
```

Comment:
Not sure how I feel about throwing errors here, those will just show up as 🔥 to users if it ever happens. Maybe silently truncating the result or showing an actual message to the user would be better (or just fall back to sending the original link).

Reply:
Ah true, I agree that sending the link makes more sense